test/dynamo/test_modules.py

```diff
@@ -2271,6 +2273,26 @@ def generate(x, c):
         generate(torch.randn(10, 10), 0)
         self.assertEqual(cnt.frame_count, 3)
 
+    def test_builtin_module_forward_always_compiled(self):
+        # module.forward should not be skipped even the higher level function call got skipped.
+        # https://github.com/pytorch/pytorch/issues/111966
+        cnt = torch._dynamo.testing.CompileCounter()
+
+        def forward_pre_hook_with_graph_break(module: torch.nn.Module, inp: Tuple):
+            torch._dynamo.graph_break()
```

Comment:
Could you perhaps add some tensor op before this to ensure the code before it will compile? 

e.g. inp[0] += 1